src/components/DropDown.tsx

```diff
@@ -1,72 +1,55 @@
 import * as React from "react";
 import { DropdownProps, DropdownState } from "./typings/Dropdown";
 import { DropDownButton } from "./Button";
-import { dropDownStyle } from "./styles/Dropdown.styles";
 import { cx } from "emotion";
-import OutsideClick from "./OutsideClick";
+import Popper from "./Popper";
 
 class DropDown extends React.PureComponent<DropdownProps, DropdownState> {
-  state: DropdownState = {
-    isOpen: this.props.initiallyOpen
-  };
-
   static defaultProps: Partial<DropdownProps> = {
     closeOnOutsideClick: true
   };
 
-  private toggleDropdown = () => {
-    this.setState({
-      isOpen: !this.state.isOpen
-    });
-  };
-
   render() {
     const {
       buttonLabel,
       children,
       labelComponent,
       padding,
-      className,
       dropDownClassName,
       isSelected,
       disabled,
-      labelClassName
+      labelClassName,
+      initiallyOpen
     } = this.props;
-    const { isOpen } = this.state;
 
     return (
-      <OutsideClick
-        onOutsideClick={() =>
-          this.setState({
-            isOpen: false
-          })
-        }
-        disabled={!isOpen}
+      <Popper
+        isOpen={initiallyOpen}
+        placement="bottom"
+        label={({ isOpen, toggle }) => (
+          <React.Fragment>
+            {labelComponent ? (
+              labelComponent({ isOpen, toggleDropdown: toggle })
+            ) : (
+              <DropDownButton
+                isSelected={isSelected}
+                isOpen={isOpen}
+                onClick={toggle}
+                disabled={disabled}
+                className={labelClassName}
+              >
+                {buttonLabel}
+              </DropDownButton>
+            )}
+          </React.Fragment>
```

Comment:
why wrapped in React.Fragment?

Reply:
Popper's label has return type defined as `JSX.Element`

Reply:
Then change the typing. 😞